src/frontend/apps/impress/src/core/AppProvider.tsx

```diff
@@ -36,6 +38,16 @@ export function AppProvider({ children }: { children: React.ReactNode }) {
     return cleanupResizeListener;
   }, [initializeResizeListener]);
 
+  useEffect(() => {
+    if (userData?.language) {
+      Object.keys(LANGUAGES_ALLOWED).forEach((key) => {
+        if (userData.language.includes(key) && i18n.language !== key) {
+          void i18n.changeLanguage(key);
+        }
+      });
+    }
+  }, [userData?.language]);
+
```

Comment:
If you rebase from main, you will see, we have now a component 
`ConfigProvider`, better to put it here, I think you will be able to get the `LANGUAGES_ALLOWED` from the conf endpoint, you will see.
https://github.com/numerique-gouv/impress/blob/main/src/frontend/apps/impress/src/core/config/ConfigProvider.tsx